superset/connectors/sqla/models.py

```diff
@@ -1834,11 +1835,20 @@ def update_table(  # pylint: disable=unused-argument
 
         session.execute(update(SqlaTable).where(SqlaTable.id == target.table.id))
 
-        # update ``Column`` model as well
         dataset = (
-            session.query(NewDataset).filter_by(sqlatable_id=target.table.id).one()
```

Comment:
This is where the error was found, but I updated all of the instances where the dataset should be updated in this file.